mrp_work_center_component/models/mrp_production.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2025 Angel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MrpProduction(models.Model):
+    _inherit = "mrp.production"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        productions = super().create(vals_list)
+        productions._add_workcenter_components()
+        return productions
+
+    def _add_workcenter_components(self):
+        components = self.env["mrp.workcenter.component"].search(
+            [("work_center_id", "in", self.workorder_ids.mapped("workcenter_id").ids)]
+        )
+        for component in components:
+            self.env["stock.move"].create(
```

Comment:
create all in the same moment, no 1 by 1. store all in one dict and after the for create all records.